packages/monorepo-scripts/src/convert_changelogs.ts

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env node
+
+import * as fs from 'fs';
+import lernaGetPackages = require('lerna-get-packages');
+import * as _ from 'lodash';
+import * as moment from 'moment';
+import * as path from 'path';
+
+import { Changelog, Changes, UpdatedPackage } from './types';
+import { utils } from './utils';
+
+const MONOREPO_ROOT_PATH = path.join(__dirname, '../../..');
+
+(async () => {
+    const allLernaPackages = lernaGetPackages(MONOREPO_ROOT_PATH);
+    const publicLernaPackages = _.filter(allLernaPackages, pkg => !pkg.package.private);
+    _.each(publicLernaPackages, lernaPackage => {
+        const changelogMdIfExists = getChangelogMdIfExists(lernaPackage.package.name, lernaPackage.location);
+        if (_.isUndefined(changelogMdIfExists)) {
+            throw new Error(`${lernaPackage.package.name} should have CHANGELOG.md b/c it's public. Add one.`);
+        }
+
+        const lines = (changelogMdIfExists as any).split('\n');
+        const changelogs: Changelog[] = [];
+        let changelog: Changelog = {
+            version: '',
+            changes: [],
+        };
+        for (const line of lines) {
+            if (_.startsWith(line, '## ')) {
+                let version = line.substr(4).split(' - ')[0];
+                if (version === '0.x.x') {
+                    version = utils.getNextPatchVersion(lernaPackage.package.version);
+                }
+                const dateStr = line.split('_')[1];
+                let date;
+                if (!_.includes(dateStr, 'TBD')) {
+                    date = moment(dateStr, 'MMMM D, YYYY');
+                }
+                changelog = {
+                    version,
+                    changes: [],
+                };
+                if (!_.isUndefined(date)) {
+                    changelog.timestamp = date.unix();
+                }
+                if (!_.includes(dateStr, 'TBD')) {
+                    changelog.isPublished = true;
+                }
+                (changelogs as any).push(changelog);
+            } else if (_.includes(line, '* ')) {
+                const note = line.split('* ')[1].split(' (#')[0];
```

Comment:
Could we use regexes with capture groups instead of that magic with splits and magic numbers?
Or at least provide an example format of the data being parsed above in a comment. 
` * Some note (#123)`